src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DoNotUseStackallocInLoopsTests.cs

```diff
@@ -306,5 +307,35 @@ public static void Foo()
                 }"
             }.RunAsync();
         }
+
+        [Fact, WorkItem(6723, "https://github.com/dotnet/roslyn-analyzers/issues/6723")]
+        public Task NoDiagnostics_StackallocInLoopInitializer()
+        {
+            return VerifyCS.VerifyAnalyzerAsync(@"
+                using System;
+                public class C
+                {
+                    public static void Foo()
+                    {
+                        for (Span<int> sp1 = stackalloc int[2]; false;) { }
```

Comment:
Can you also add tests that show:
- It still triggers when the stackalloc is in the condition or update expressions
- It doesn't trigger when a variable other than the first contains the stackalloc
- It doesn't trigger when the initializer contains the stackalloc as a subexpression, eg `Span<int> sp1 = Helper(stackalloc int[2])`

Reply:
Sure! Done.